src/GAP/iso_gap_oscar.jl

```diff
@@ -0,0 +1,192 @@
+
+   return MapFromFunc(f, finv, F, G)
+end
+
+# For finite non-prime fields, choose `FiniteField(pol, "z")`
+# (of type `FqNmodFiniteField` if `p` is small, `FqFiniteField` otherwise)
+# because we can choose the same polynomial as is used in GAP.
+function _iso_gap_oscar_field_finite_ext(F::GAP.GapObj)
+   p = characteristic(F)  # of type `fmpz`
+   if p < fmpz(2)^64
+     p = UInt64(p)
+   end
+   d = GAP.Globals.DegreeOverPrimeField(F)
+   e = GAPWrap.One(F)
+
+   R, x = PolynomialRing(GF(p), "x")
+   if GAP.Globals.IsPrimeField(GAP.Globals.LeftActingDomain(F))
+     pol_gap = GAP.Globals.DefiningPolynomial(F)
+     if GAPWrap.IsAlgebraicExtension(F)
+#FIXME:
+# As soon as the problem from https://github.com/gap-system/gap/issues/4694
+# is fixed, go back to `Basis_F = GAP.Globals.Basis(F)` also in this case.
+# Note that we want a basis in the filter `IsCanonicalBasisAlgebraicExtension`.
+        Basis_F = GAP.Globals.Objectify(GAP.Globals.NewType(
+                  GAP.Globals.FamilyObj(F),
+                  GAP.Globals.IsCanonicalBasisAlgebraicExtension),
+                  GAP.NewPrecord(0))
+        GAP.Globals.SetUnderlyingLeftModule(Basis_F, F)
+     else
+        Basis_F = GAP.Globals.Basis(F)
+     end
+   else
+     # The GAP field is not an extension of the prime field.
+     # What is a reasonable way to compute (on the GAP side) a polynomial
+     # w.r.t. the prime field, and to decompose field elements w.r.t.
+     # the corresponding basis?
+     error("extensions of extension fields are not supported")
+   end
+
+   coeffs = GAP.Globals.CoefficientsOfUnivariatePolynomial(pol_gap)
+   coeffs = [fmpz(Oscar.GAPWrap.IntFFE(x)) for x in coeffs]
+   pol = R(coeffs)
+   G = FiniteField(pol, "z")[1]
+   Basis_G = Vector{elem_type(G)}(undef, d)
+   Basis_G[1] = G(1)
+   for i = 2:d
+      Basis_G[i] = Basis_G[i - 1]*gen(G)
+   end
+
+   function f(x)
+      v = GAPWrap.Coefficients(Basis_F, x)
+      v_int = [fmpz(GAPWrap.IntFFE(v[i])) for i = 1:length(v)]
+      return sum([v_int[i]*Basis_G[i] for i = 1:d])
+   end
+
+   function finv(x::Union{fq_nmod, fq})
+      v = [GAP.Obj(coeff(x, i)) for i in 0:(d - 1)]
+      return sum([v[i]*Basis_F[i] for i in 1:d])
+   end
+
+   return MapFromFunc(f, finv, F, G)
+end
+
+function _iso_gap_oscar_field_rationals(F::GAP.GapObj)
+   return MapFromFunc(x -> fmpq(x), x -> GAP.Obj(x), F, QQ)
+end
+
+function _iso_gap_oscar_field_cyclotomic(F::GAP.GapObj)
+   N = GAPWrap.Conductor(F)
+   cycpol = GAP.Globals.CyclotomicPol(N)
+   dim = length(cycpol)-1
+   G = CyclotomicField(N)[1]
+
+   function f(x)
+      GAPWrap.IsCyc(x) || error("$x is not a GAP cyclotomic")
+      denom = GAPWrap.DenominatorCyc(x)
+      n = GAPWrap.Conductor(x)
+      mod(N, n) == 0 || error("$x does not lie in the $N-th cyclotomic field")
+      coeffs = GAP.Globals.CoeffsCyc(x * denom, N)
+      GAP.Globals.ReduceCoeffs(coeffs, cycpol)
+      coeffs = Vector{fmpz}(coeffs)
+      coeffs = coeffs[1:dim]
```

Comment:
Just out of curiosity, why is this line necessary?

I also note that we have essentially the same code now in three places

Reply:
`GAP.Globals.ReduceCoeffs` does not remove trailing zeroes, and we need a vector of length `dim`.

Concerning the code duplication: Yes. I had already thought about using the inverse of the `iso_oscar_gap` mapping as `iso_gap_oscar` where this is possible.